services/proxy/pkg/command/server.go

```diff
@@ -137,7 +188,71 @@ func Server(cfg *config.Config) *cli.Command {
 	}
 }
 
-func loadMiddlewares(ctx context.Context, logger log.Logger, cfg *config.Config) alice.Chain {
+func (h *StaticRouteHandler) handler() http.Handler {
+	m := chi.NewMux()
+	var methods = []string{"PROPFIND", "DELETE", "PROPPATCH", "MKCOL", "COPY", "MOVE", "LOCK", "UNLOCK", "REPORT"}
+	for _, k := range methods {
+		chi.RegisterMethod(k)
```

Comment:
Hm, this rings a bell. chi.RegisterMethod(k) is racey as it writes to a global map, which other go-routines might be accessing.

We added workarounds for this in other places. See e.g.: https://github.com/cs3org/reva/pull/2785/files and https://github.com/owncloud/ocis/pull/3662